src/processes.c

```diff
@@ -2694,33 +2561,11 @@ static int ps_read (void)
 		pse.id = pid;
 		pse.age = 0;
 
-		pse.num_proc   = ps.num_proc;
-		pse.num_lwp    = ps.num_lwp;
-		pse.vmem_size  = ps.vmem_size;
-		pse.vmem_rss   = ps.vmem_rss;
-		pse.vmem_data  = ps.vmem_data;
-		pse.vmem_code  = ps.vmem_code;
-		pse.stack_size = ps.stack_size;
-
-		pse.vmem_minflt = 0;
-		pse.vmem_minflt_counter = ps.vmem_minflt_counter;
-		pse.vmem_majflt = 0;
-		pse.vmem_majflt_counter = ps.vmem_majflt_counter;
-
-		pse.cpu_user = 0;
-		pse.cpu_user_counter = ps.cpu_user_counter;
-		pse.cpu_system = 0;
-		pse.cpu_system_counter = ps.cpu_system_counter;
-
-		pse.io_rchar = ps.io_rchar;
-		pse.io_wchar = ps.io_wchar;
-		pse.io_syscr = ps.io_syscr;
-		pse.io_syscw = ps.io_syscw;
-		pse.io_diskr = ps.io_diskr;
-		pse.io_diskw = ps.io_diskw;
-
-		pse.cswitch_vol = -1;
-		pse.cswitch_invol = -1;
+		pse.gauges = ps.gauges;
+		pse.counters = ps.counters;
+
+		pse.gauges.cswitch_vol = -1;
```

Comment:
This already happens in `ps_read_process`, so the above `pse.gauges = ps.gauges` assignment should take care of this.

Reply:
This is only true if report_ctx_switch is enabled, and some error is returned. The fact that it was set intentionally before has me concerned that it should still be set intentionally.

Reply:
This is the Solaris case -- if you look at `ps_read_process` for Solaris, there the assignment happens unconditionally.

Reply:
Done.